### PR TITLE
Rework the aide test case and fix the initialized database error

### PR DESCRIPTION
**Pull request by jouyingbin**
1. Rework and Fix the aide initialize database error
2. Add more check on differences between datebase and file system

- Related ticket: https://progress.opensuse.org/issues/64364
- Needles: NA
- Verification run: http://10.163.2.52/tests/558

### CODE REVIEW

File: tests/console/aide_check.pm [hunk 1]
```diff
@@ -1,42 +1,73 @@
...
+    send_key 'ret';
+
     assert_script_run "touch /var/log/testlog";
-    assert_script_run "clear";
-    validate_script_output "aide --check || true", sub { m/found differences between database and filesystem/ }, 60;
+
+    send_key 'ret';
+
+    # Checks the database for inconsistencies and there is 2 new added entry
+    validate_script_output "aide --check || true", sub { m/AIDE found differences between database and filesystem/ && m/Added entries:(\s+)2/ }, 300;
```
Comment (lilyeyes) [inline tests/console/aide_check.pm:59]: why is "||" needed?

File: tests/console/aide_check.pm [hunk 2]
```diff
@@ -1,42 +1,73 @@
...
     zypper_call "in aide";
+
     assert_script_run "cp /etc/aide.conf /etc/aide.conf.bak";
-    assert_script_run "sed -i 's:^/:!/:g' /etc/aide.conf && sed -i 's:!/var/log:/var/log:g' /etc/aide.conf";
-    assert_script_run "aide -i", 60;
+
+    # Initialize the database and move it to the appropriate place before using the --check command
+    validate_script_output "aide --init || true", sub { m/AIDE initialized database/ }, 300;
+
+    send_key 'ret';
```
Comment (coolgw) [inline tests/console/aide_check.pm:45]: why u need send 'ret'?

Comment (jouyingbin) [reply tests/console/aide_check.pm:45]: The output message will shift to the rightside after running the aide --init, it is more readable after sending 'ret' in the verification run.

File: tests/console/aide_check.pm (hunk 1 above)
Comment (jouyingbin) [reply tests/console/aide_check.pm:59]: I think it is the code style question :), I think these 2 methods will both work well.
`"aide --check || true"`  and `"aide --check" ` . It is no harm to keep it.

**Comment by lilyeyes** [comment]
> I think it is the code style question :), I think these 2 methods will both work well.
> "aide --check || true" and "aide --check" . It is no harm to keep it.

They are different.
`$cmd1 || $cmd2` means if $cmd1 failed then execute $cmd2.
So,  `aide --check || true` means if `aide --check` failed then execute `true`. the returen value is always 0 (true).

**Comment by jouyingbin** [comment]
> > I think it is the code style question :), I think these 2 methods will both work well.
> > "aide --check || true" and "aide --check" . It is no harm to keep it.
> 
> They are different.
> `$cmd1 || $cmd2` means if $cmd1 failed then execute $cmd2.
> So, `aide --check || true` means if `aide --check` failed then execute `true`. So the return value is always 0 (true).

Yes, you are right. I just do some experimental test. The return is always 0. I will modify it again. Thanks.

**Comment by jouyingbin** [comment]
> > I think it is the code style question :), I think these 2 methods will both work well.
> > "aide --check || true" and "aide --check" . It is no harm to keep it.
> 
> They are different.
> `$cmd1 || $cmd2` means if $cmd1 failed then execute $cmd2.
> So, `aide --check || true` means if `aide --check` failed then execute `true`. So the return value is always 0 (true).

Yes, you are right. I just do some experimental test. The return is always 0. I will modify it again. Thanks.

**Pull request closed without merging**